ddtrace/debugging/_config.py

```diff
@@ -0,0 +1,101 @@
+import os
+from typing import Dict
+from typing import Optional
+
+from ddtrace import config as tracer_config
+from ddtrace.internal.constants import DEFAULT_SERVICE_NAME
+from ddtrace.internal.logger import get_logger
+from ddtrace.internal.utils.config import get_application_name
+from ddtrace.internal.utils.formats import asbool
+from ddtrace.version import get_version
+
+
+log = get_logger(__name__)
+
+
+DEFAULT_DEBUGGER_PORT = 8126
+DEFAULT_PROBE_API_URL = "https://app.datadoghq.com"
+DEFAULT_SNAPSHOT_INTAKE_URL = "http://%s:%d" % (os.getenv("DD_AGENT_HOST", "localhost"), DEFAULT_DEBUGGER_PORT)
+DEFAULT_MAX_PROBES = 100
+DEFAULT_METRICS = True
+DEFAULT_PROBE_STATUS_URL = DEFAULT_SNAPSHOT_INTAKE_URL
+DEFAULT_GLOBAL_RATE_LIMIT = 100.0
+DEFAULT_AGENT_MODE = False
+DEFAULT_API_KEY_FILE = "/etc/datadog/secrets/datadog-api-key"
+
+
+def _get_api_key_from_file():
+    # type: () -> Optional[str]
+    """Retrieve the API key from file."""
+    try:
+        key_file = os.getenv("DD_API_KEY_FILE", DEFAULT_API_KEY_FILE)
+        with open(key_file) as _:
+            return _.read().strip()
+    except (IOError, OSError):
+        return None
+
+
+SECRETS = frozenset(["api_key"])
+
+
+class DebuggerConfig(object):
+    """Debugger configuration."""
+
+    api_key = None  # type: Optional[str]
+    service_name = DEFAULT_SERVICE_NAME
+    probe_api_url = DEFAULT_PROBE_API_URL
+    snapshot_intake_url = DEFAULT_SNAPSHOT_INTAKE_URL
+    probe_status_url = DEFAULT_PROBE_STATUS_URL
+    max_probes = DEFAULT_MAX_PROBES
+    metrics = DEFAULT_METRICS
+    global_rate_limit = DEFAULT_GLOBAL_RATE_LIMIT
+    tags = None  # type: Optional[str]
+    _tags = {}  # type: Dict[str, str]
+    _tags_in_qs = True
+    _snapshot_intake_endpoint = "/v1/input"
+    _probe_status_endpoint = _snapshot_intake_endpoint
+    _agent = DEFAULT_AGENT_MODE
+
+    def __init__(self):
+        # type: () -> None
+        api_key = os.getenv("DD_API_KEY", None)
```

Comment:
Good point. I think this can be considered legacy from an earlier implementation so perhaps we can do away with this as we would rely on an agent for this. I'll get rid of this for now, and if we really need it we can put it back (but I doubt it'll be the case!)